### PR TITLE
switch to archive data with serde serialisation

### PR DESCRIPTION
**Pull request by AnderEnder**

### CODE REVIEW

File: whisper/src/bin/whisper-fetch.rs [hunk 1]
```diff
@@ -87,33 +84,31 @@ fn main() -> Result<(), Error> {
...
-            "end": archive.until_interval,
-            "step": archive.step,
-            "values": values,
-            });
-
-        println!("{}", serde_json::to_string_pretty(&john)?);
+        println!("{}", serde_json::to_string_pretty(&archive)?);
     } else {
-        for (index, value) in values.iter().enumerate() {
+        for (index, value) in archive.values.iter().enumerate() {
```
Comment (andy128k) [inline whisper/src/bin/whisper-fetch.rs]: This loop looks sub-optimal.

File: whisper/src/bin/whisper-fetch.rs [hunk 2]
```diff
@@ -87,33 +84,31 @@ fn main() -> Result<(), Error> {
...
 
-    let values: Vec<String> = archive
-        .values
-        .into_iter()
-        .filter(&filter)
-        .map(|v| v.map(|x| x.to_string()).unwrap_or("None".to_owned()))
-        .collect();
+    let archive = file.fetch(seconds_per_point, interval, now)?
+        .ok_or(err_msg("No data in selected timerange"))?
+        .filter_out(&filter);
```
Comment (loony-bean) [inline whisper/src/bin/whisper-fetch.rs]: filter_map?
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map

Comment (AnderEnder) [reply whisper/src/bin/whisper-fetch.rs]: For what purposes? Filter is inside the structure, like @andy128k proposed to do in previous PR review, so filter_out returns the same struct ArchiveData.

File: whisper/src/bin/whisper-fetch.rs (hunk 1 above)
Comment (AnderEnder) [reply whisper/src/bin/whisper-fetch.rs]: I am not sure I understand you :)
Any details or alternatives?

File: whisper/src/bin/whisper-fetch.rs [hunk 3]
```diff
@@ -87,36 +83,43 @@ fn main() -> Result<(), Error> {
...
-            "values": values,
-            });
-
-        println!("{}", serde_json::to_string_pretty(&john)?);
+        println!("{}", serde_json::to_string_pretty(&archive)?);
     } else {
-        for (index, value) in values.iter().enumerate() {
+        let time_format = match (&args.pretty, &args.time_format) {
+            (true, Some(time_format)) => Some(time_format),
+            (_, _) => None,
```
Comment (andy128k) [inline whisper/src/bin/whisper-fetch.rs]: just _

**Pull request merged**